### ISSUE TITLE
[Spell] Find Familiar

### ISSUE THREAD

**Issue by FelipeRenault** [ZerdBG3/DnD-Rebalancing#80]
- Remove attacks from normal Familiars
- Develop action for Pact of the Chain Warlocks provide attacks to their familiars by using the Warlock's action
- Include Help (Distract) action for familiars when #5 is done

**Comment by FelipeRenault** [comment]
Find Familiar is an 1-hour cast and have component cost, which isn't implemented yet. It also has the ritual tag, so my idea is to make the spell only be used outside of combat, but not consume a spell slot (no one would ever use the traditional spell slot for an 1-hour cast instead of 1h10 min for no spell slot). It also has the problem that rituals (for wizards, at least) don't need to be prepared to be cast as rituals if they have it on their spellbook, which I have no idea how to implement. So here's the possible extra changes I have in mind, which might not be doable

- Out-of-combat only, but doesn't cost a spell slot (doable)
- Costs an specific item (or 10gp) to cast (not sure if doable)
- Usable from spellbook without preparing for wizards (probably not doable - which isn't a huge problem, since currently, preparing is done at any time outside of combat, not only at long rests)
- Dismiss action, which should give the recast action for free for it to reappear (and can be used in-combat)
- Touch-spell delivery system (I'd rather develop my own game than try to implement this on BG3)

**Comment by gnufred** [comment]
I'd stay, out of combat only. It's a good first step. The 10GP would be closer to RAW but familiars aren't as much nearly as helpful in BG3 than on the Tabletop.

And if we think about ritual spells in general, out-of-combat free cast is pretty much the simplest best we can get. If we can make the casting time a bit longer (like 2x or 3x) that would be closer to RAW (but 10 minute might not work here).

**Comment by FelipeRenault** [comment]
I don't think we can increase cast time. There is no cast time bigger than 1 action in BG3. So we either have in-combat or out-of-combat.
If the usual cast time is 1 action or 1 bonus action and it has the ritual tag, then we get a clear difference between a regular spell that can be used in combat and the ritual that can only be used outside of combat. For spells like Find Familiar, though, the regular use already requires an only out-of-comba situation, since it's an 1 hour cast time. In that case, there's no conceivable situation (in BG3) that someone would want to use a spell slot, so I'll make it the ritual version only.

**Comment by gnufred** [comment]
Sounds like we have a pretty solid solution for Find Familiar (and other ritual spells).

**Comment by FelipeRenault** [comment]
Are there any other Ritual spells currently in BG3?

**Comment by gnufred** [comment]
Good question! I checked the list from https://baldursgate3.wiki.fextralife.com/Spells and from http://dnd5e.wikidot.com/spells:ritual and the answer would be: none. So, only find familiar.

**Comment by FelipeRenault** [comment]
Update: now there is a `RitualCost` option for spells, which make them have different costs inside or outside of combat. This eliminates the need of having separate spells with separate costs.
Also #125 makes a very good progress on making Find Familiars _almost_ work properly. I just need to adjust the ones from Pact of the Chain, possibly giving the caster something along the lines of Commander Strike

**Comment by drkekyll** [comment]
Not sure if you're done with Find Familiar yet, but if you wanted to simulate a longer casting time, you could apply a status to the caster that blocks action resources for a duration.

**Comment by FelipeRenault** [comment]
I think just making it Out of Combat is enough. I don't see any benefit on making the player wait for a spell to be cast while they're not time pressured, anyway. Just kills the flow of the game

**Comment by drkekyll** [comment]
Oh I agree. Just figured it was worth mentioning.
